### PR TITLE
Show more than one group color bar

### PR DESCRIPTION
**Pull request by tobiasdiez**
Shows multiple colored bars if the entry belongs to more than one group (with color). Implements #4574.
image

----

- [ ] Change in CHANGELOG.md described
- [ ] Tests created for changes
- [ ] Manually tested changed features in running JabRef
- [ ] Screenshots added in PR description (for bigger UI changes)
- [ ] Ensured that [the git commit message is a good one](https://github.com/joelparkerhenderson/git_commit_message)
- [ ] Check documentation status (Issue created for outdated help page at [help.jabref.org](https://github.com/JabRef/help.jabref.org/issues)?)

### CODE REVIEW

File: src/main/java/org/jabref/gui/maintable/MainTableColumnFactory.java [hunk 1]
```diff
@@ -129,27 +129,26 @@ public MainTableColumnFactory(BibDatabaseContext database, ColumnPreferences pre
...
-            StackPane container = new StackPane();
+        List<Color> groupColors = matchedGroups.stream()
+                                               .flatMap(group -> OptionalUtil.toStream(group.getColor()))
+                                               .collect(Collectors.toList());
+
+        if (!groupColors.isEmpty()) {
+            HBox container = new HBox();
+            container.setSpacing(2);
             container.setMinWidth(10);
             container.setAlignment(Pos.CENTER);
```
Comment (matthiasgeiger) [inline src/main/java/org/jabref/gui/maintable/MainTableColumnFactory.java]: Is it possible to set it to left align? Looks better imho 😀

Comment (tobiasdiez) [reply src/main/java/org/jabref/gui/maintable/MainTableColumnFactory.java]: Good idea, but now I cannot decide what I prefer 😸 

Left-aligned
image

Right-aligned
image

or the centered version from above...?!

Comment (matthiasgeiger) [reply src/main/java/org/jabref/gui/maintable/MainTableColumnFactory.java]: I would still say left aligned... But that's just my personal opinion 😁

**Comment by koppor** [comment]
+1 for left-aligned. If one does not know the feature, this looks better to me.

Anything else left?

**Comment by tobiasdiez** [comment]
ok, left aligned it is. Ready to merge from my POV.

**Pull request merged**